purview/api/v1.py

```diff
@@ -17,3 +20,15 @@ def about():
     :return: A JSON object with version info
     """
     return jsonify({'version': version})
+
+
+@api_v1.route('/<resource>/<uid>')
+def get_resource(resource, uid):
+    for model in all_models:
+        if model.__name__.lower() == resource:
+            for prop_name, prop_def in model.defined_properties().items():
+                if isinstance(prop_def, UniqueIdProperty):
+                    item = model.nodes.get(**{prop_def.name: uid})
+                    if item:
+                        return jsonify(item.serialized)
+    raise NotFound('This item does not exist')
```

Comment:
After thinking about this exception, I think should this should be in an else statement under `if item:` and then this line should have a different exception letting the user know they request an invalid resource. Something like:

```python
# TODO: This variable should be defined above and then the for loop could use this as well
model_names = [model.__name__.lower() for model in all_models]
error = 'The requested resource "{0}" is invalid. Choose from the following: {1}, and {2}.'.format(
    resource, ', '.join(model_names[:-1]), model_names[-1])
raise ValidationError()